core/task_manager.py

```diff
@@ -98,13 +98,13 @@ def stop_threads(self):
             # To be improved with a full async implementation
             # To be done : "asyncio.run" --> replaced by a simple await
             # PR discussion : https://github.com/Drakkar-Software/OctoBot/pull/563#discussion_r248088266
-            stop_coroutines.append(
-                self.octobot.get_config()[CONFIG_NOTIFICATION_INSTANCE].notify_with_all(NOTIFICATION_STOPPING_MESSAGE))
+            stop_coroutines.append(self.octobot.get_config()[CONFIG_NOTIFICATION_INSTANCE]
+                                   .notify_with_all(NOTIFICATION_STOPPING_MESSAGE))
 
         self.logger.info("Stopping threads ...")
 
         if self.main_task_group:
-            self.main_task_group.cancel()
+            self.async_loop.call_soon_threadsafe(self.main_task_group.cancel)
```

Comment:
https://stackoverflow.com/questions/29296064/python-asyncio-how-to-create-and-cancel-tasks-from-another-thread